server.go

```diff
@@ -589,14 +575,12 @@ func statusFromError(p ider, err error) sshFxpStatusPacket {
 	ret.StatusError.Code = sshFxFailure
 	ret.StatusError.msg = err.Error()
 
+	if code, ok := translateSyscallError(err); ok {
```

Comment:
From the other PR, for better Windows compat of a file underneath a non-existing directory, we probably want to switch to using `if os.IsNotExist(err) { ret.StatusError.Code = sshFxNoSuchFile ; return ret }` which should simplify up a lot of your code handling here? As after, we would not need any special syscall error code handling for file-not-found errors.

Reply:
Yeah, it's better to use `os.IsNotExist` and `os.Permission` here. I was trying to keep the changes to minimum, but I can certainly make that change.

Reply:
We still need syscall error code handing for the case when `errno == 0` (we return `sshFxOk`), and when errno is non-zero but not `ENOENT` or `EPERM`, in which case we return `sshFxFailure`.

I'm inclined to leave this change out of this PR, since we're just trying to fix build on plan9 (and not break other systems in the process).

Reply:
I’m going to recommend that you hold off of that change. We’ll probably merge the other PR before yours, and then you’ll have a merge conflict to clean up, but hopefully it shouldn’t be too bad. Especially, since their code only touches a strict subset of your code, so you should be pretty good and familiar with the code to make the right conflict resolution.

Reply:
The other PR has been merged, and as expected, github now reports there are merge conflicts to resolve. :smiley: 

Once you‘ve addressed those, I think we‘re good to commit. Thanks for the work! I know it can be hard to find people who can even test on plan9, let alone do any support work necessary.

Reply:
Done. Thanks for the review!